week_2/bus_stops_3/bus_stops_3.cpp

```diff
@@ -0,0 +1,34 @@
+#include <iostream>
+#include <map>
+#include <set>
+#include <string>
+
+int main() {
+  int Q;
+  std::cin >> Q;
+  std::string stop;
+  int stops_count;
+  int k = 1;
+  std::set<std::string> stops;
+  std::map<std::set<std::string>, int> route;
+
+  for (int it = 0; it < Q; ++it) {
+    std::cin >> stops_count;
+
+    for (int i = 0; i < stops_count; ++i) {
+      std::cin >> stop;
+      stops.insert(stop);
+    }
+
+    if (route.count(stops) != 0) {
+      std::cout << "Already exists for " << route[stops] << std::endl;
+    } else {
+      route[stops] = k;
+      std::cout << "New bus " << route[stops] << std::endl;
+      ++k;
+    }
+    stops.clear();
```

Comment:
@Ypostolov Why should we clear stops ? What if remove this line?

Reply:
To have possibility to use this var for different routes.